cypress/e2e/functional/tile_tests/geometry_tool_spec.js

```diff
@@ -277,6 +356,164 @@ context('Geometry Tool', function () {
+    });
+
+    // Add an angle label
+    geometryToolTile.getGraphPointLabel().contains('°').should('not.exist');
+    clueCanvas.clickToolbarButton('geometry', 'select');
+    geometryToolTile.selectGraphPoint(20, 5); // this point is a 45-degree angle but sometimes it starts at 44
+    clueCanvas.clickToolbarButton('geometry', 'label');
+    geometryToolTile.toggleAngleCheckbox();
+
+    // Use a flexible check to verify either '44°' or '45°'
+    cy.log('Verify the initial angle is either 44° or 45°');
+    geometryToolTile.getGraphPointLabel().then(($label) => {
+      const angleText = $label.text();
+      if (angleText.includes('44°')) {
+        cy.wrap(44).as('initialAngle');
+      } else if (angleText.includes('45°')) {
+        cy.wrap(45).as('initialAngle');
+      } else {
+        throw new Error(`Unexpected angle: ${angleText}`);
+      }
+    });
+
+    // Add length labels to two line segments
+    // Select line segments by clicking between two points
+    geometryToolTile.clickGraphPosition(7.5, 5); // Middle of the first segment between (5, 5) and (10, 5)
+    clueCanvas.clickToolbarButton('geometry', 'label');
+    geometryToolTile.getModalTitle().should('contain.text', 'Length');
+    geometryToolTile.chooseLabelOption('length');
+
+    geometryToolTile.clickGraphPosition(15, 7.5); // Middle of the second segment between (15, 5) and (15, 10)
+    clueCanvas.clickToolbarButton('geometry', 'label');
+    geometryToolTile.getModalTitle().should('contain.text', 'Length');
+    geometryToolTile.chooseLabelOption('length');
+    geometryToolTile.clickGraphPosition(0, 0); // deselect
+
+    // Verify that the two line segments were created
+    geometryToolTile.getGraphPointLabel().contains('10.0').should('exist');
+    geometryToolTile.getGraphPointLabel().contains('5.0').should('exist');
+
+    // Move the point
+    clueCanvas.clickToolbarButton('geometry', 'select');
+    geometryToolTile.clickGraphPosition(15, 5); // shared point
+    geometryToolTile.getSelectedGraphPoint().trigger('keydown', { keyCode: 39 }); // simulate right arrow key press
+    geometryToolTile.getSelectedGraphPoint().trigger('keydown', { keyCode: 38 }); // simulate up arrow key press
+
+    // Verify that the point values changed
+    cy.get('@originalCx').then((originalCx) => {
+      cy.get('@originalCy').then((originalCy) => {
+        geometryToolTile.getSelectedGraphPoint().then(($point) => {
+          const newCx = parseFloat($point.attr('cx'));
+          const newCy = parseFloat($point.attr('cy'));
+
+          expect(newCx).to.be.greaterThan(originalCx);
+          expect(newCy).to.be.lessThan(originalCy);
+        });
+      });
+    });
+
+    // Verify that the angle has decreased by 1 degree
+    cy.get('@initialAngle').then((initialAngle) => {
+      const newAngle = initialAngle - 1;
+      geometryToolTile.getGraphPointLabel().contains(`${newAngle}°`).should('exist'); // Check if angle decreased by 1 degree
```

Comment:
As above for cx & cy, you could change this to check that the value is less, rather than a specific value; however, if you do, you would need to parse out (i.e. remove) the degree sign (°) from the new value returned by getGraphPointLabel(), so that you can compare integers.